app/controllers/conversations_controller.rb

```diff
@@ -32,4 +40,16 @@ def conversation
   def conversation_params
     params.require(:conversation).permit(:target_id, :initiator_id)
   end
+
+  def conversation_update_params
+    params.require(:conversation).permit(:unread)
+  end
+
+  def update_params
+    @update_params ||= if conversation.messages.last&.user == current_user
+                         conversation_update_params.except :unread
```

Comment:
So... no params here? Am I reading this correctly? Since you are only permitting `unread` param

Reply:
Can I set unread to true??

Reply:
Yep, you could. Only if you are no the last user to send a message in the conversation. 

This `except` here is to avoid the possibility of the last user updating the conversation to be marked as read (since he is the very last person to send a message he shouldn't be able to update that attribute) and hence making the other user of the conversation change his unread status unexpectedly. 

Is it too messy?

Reply:
No, just wondering if this was the best approach... so you want to be able to change a conversation to `unread: true` from outside? seems weird don't you think?

Reply:
Ahmm I thought of this `update` action as a generic "restful-resource" update.

So if you want to mark the conversation as unread, just send the params with unread set as false. 

If somehow the client ends up needing to mark it as true, you also can. 
But we also make sure that the unread attribute is only modifiable by the second last user to send a message. 

Sounds tricky,  maybe it would be more readable if we stop using `unread` as a column in conversations and instead use an array of unread conversations in the users column, or something different than handling the unread status for both users in one attribute. Idk.